php-frontend/src/main/java/org/sonar/php/parser/NewPHPGrammar.java

```diff
@@ -0,0 +1,1539 @@
+      .is(f.catchBlock(
+        b.token(PHPKeyword.CATCH),
+        b.token(PHPPunctuator.LPARENTHESIS),
+        NAMESPACE_NAME(),
+        b.token(PHPLexicalGrammar.REGULAR_VAR_IDENTIFIER),
+        b.token(PHPPunctuator.RPARENTHESIS),
+        BLOCK()
+      ));
+
+  }
+
+  public BlockTree BLOCK() {
+    return b.<BlockTree>nonterminal(PHPLexicalGrammar.BLOCK)
+      .is(f.block(
+        b.token(PHPPunctuator.LCURLYBRACE),
+        b.zeroOrMore(INNER_STATEMENT()),
+        b.token(PHPPunctuator.RCURLYBRACE)
+      ));
+  }
+
+  public GotoStatementTree GOTO_STATEMENT() {
+    return b.<GotoStatementTree>nonterminal(PHPLexicalGrammar.GOTO_STATEMENT)
+      .is(f.gotoStatement(b.token(PHPKeyword.GOTO), b.token(PHPLexicalGrammar.IDENTIFIER), EOS()));
+  }
+
+  public ExpressionStatementTree EXPRESSION_STATEMENT() {
+    return b.<ExpressionStatementTree>nonterminal(PHPLexicalGrammar.EXPRESSION_STATEMENT)
+      .is(f.expressionStatement(EXPRESSION(), EOS()));
+  }
+
+  public LabelTree LABEL() {
+    return b.<LabelTree>nonterminal(PHPLexicalGrammar.LABEL)
+      .is(f.label(b.token(PHPLexicalGrammar.IDENTIFIER), b.token(PHPPunctuator.COLON)));
+  }
+
+  public InternalSyntaxToken EOS() {
+    return b.<InternalSyntaxToken>nonterminal(PHPLexicalGrammar.EOS)
+      .is(b.firstOf(b.token(PHPPunctuator.SEMICOLON), b.token(PHPLexicalGrammar.INLINE_HTML)));
+  }
+
+  /**
+   * [ END ] Statement
+   */
+
+  /**
+   * [ START ] Expression
+   */
+
+  public ExpressionTree CAST_EXPR() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.CAST_TYPE).is(
+      f.castExpression(
+        b.token(PHPPunctuator.LPARENTHESIS),
+        b.firstOf(
+          b.token(PHPKeyword.ARRAY),
+          b.token(PHPKeyword.UNSET),
+          b.token(PHPLexicalGrammar.INTEGER),
+          b.token(PHPLexicalGrammar.INT),
+          b.token(PHPLexicalGrammar.DOUBLE),
+          b.token(PHPLexicalGrammar.FLOAT),
+          b.token(PHPLexicalGrammar.STRING),
+          b.token(PHPLexicalGrammar.OBJECT),
+          b.token(PHPLexicalGrammar.BOOLEAN),
+          b.token(PHPLexicalGrammar.BOOL),
+          b.token(PHPLexicalGrammar.BINARY)),
+        b.token(PHPPunctuator.RPARENTHESIS),
+        UNARY_EXPR()));
+  }
+
+  public ExpressionTree UNARY_EXPR() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.UNARY_EXPR).is(
+      b.firstOf(
+        f.unaryExpr(
+          b.firstOf(
+            b.token(PHPPunctuator.INC),
+            b.token(PHPPunctuator.DEC),
+            b.token(PHPPunctuator.PLUS),
+            b.token(PHPPunctuator.MINUS),
+            b.token(PHPPunctuator.TILDA),
+            b.token(PHPPunctuator.BANG),
+            b.token(PHPPunctuator.AT)),
+          UNARY_EXPR()),
+        CAST_EXPR(),
+        // FIXME (PY) Uncomment the following lines
+        // ASSIGNMENT_EXPR(),
+        // ASSIGNMENT_BY_REFERENCE(),
+        POSTFIX_EXPRESSION()));
+  }
+
+  public ExpressionTree CONCATENATION_EXPR() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00100)